core/src/main/java/hudson/model/User.java

```diff
@@ -752,6 +752,7 @@ public synchronized void delete() throws IOException {
             byNameLock.readLock().unlock();
         }
         Util.deleteRecursive(new File(getRootDir(), strategy.filenameOf(id)));
+        UserDetailsCache.get().invalidate(strategy.keyFor(id));
```

Comment:
@rsandell This is good.  I think this should apply to rekey( ), reload( ), (which would be an invalidate all, maybe).   doConfigSubmit would be an invalidation on the old name, I think.